lib/bitcoin/lock/LockResolver.ts

```diff
@@ -0,0 +1,139 @@
+import BitcoinClient from '../BitcoinClient';
+import BitcoinError from '../BitcoinError';
+import BitcoinOutputModel from '../models/BitcoinOutputModel';
+import BlockchainLockModel from '../../common/models/BlockchainLockModel';
+import ErrorCode from '../ErrorCode';
+import LockIdentifier from '../models/LockIdentifierModel';
+import LockIdentifierSerializer from './LockIdentifierSerializer';
+import { Script, Address } from 'bitcore-lib';
+
+/**
+ * Encapsulates functionality for verifying a bitcoin lock created by this service.
+ */
+export default class LockResolver {
+
+  constructor (private bitcoinClient: BitcoinClient) {
+  }
+
+  /**
+   * Gets the corresponding lock information represented by the specified lock identifier. it also verifies
+   * the lock by making sure that the corresponding transaction is indeed a lock transaction paying to the
+   * wallet in the lockIdentifier upon lock expiry.
+   *
+   * @param lockIdentifier The lock identifier.
+   * @returns The blockchain lock model if the specified identifier is verified; throws if verification fails.
+   */
+  public async resolveLockIdentifierAndThrowOnError (lockIdentifier: LockIdentifier): Promise<BlockchainLockModel> {
+
+    // The verifictation of a lock-identifier has the following steps:
+    //   (A). The redeem script in the lock-identifier is actually a 'locking' script
+    //   (B). The redeem script in the lock-identifier is paying to the wallet in the lock-identifier
+    //   (C). The transaction in the lock-identifier is paying to the redeem script in the lock-identifier
+    //
+    // With above, we can verify that the amount is/was locked for the specified wallet in

```

Comment:
This function is not accessing the amount. It is only verifying A, B and C, how do we verify that the amount is correct and if the lock is still active or not?

Reply:
Or is it core's job to do that?

Reply:
@isaacJChen, you are right that the core will do that.